contracts/Optimism_SpokePool.sol

```diff
@@ -44,37 +47,55 @@ contract Optimism_SpokePool is CrossDomainEnabled, SpokePoolInterface, SpokePool
         public
         override
         onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
     {
         _setCrossDomainAdmin(newCrossDomainAdmin);
     }
 
+    function setHubPool(address newHubPool) public override onlyFromCrossDomainAccount(crossDomainAdmin) nonReentrant {
+        _setHubPool(newHubPool);
+    }
+
     function setEnableRoute(
         address originToken,
         uint256 destinationChainId,
         bool enable
-    ) public override onlyFromCrossDomainAccount(crossDomainAdmin) {
+    ) public override onlyFromCrossDomainAccount(crossDomainAdmin) nonReentrant {
         _setEnableRoute(originToken, destinationChainId, enable);
     }
 
-    function setDepositQuoteTimeBuffer(uint64 buffer) public override onlyFromCrossDomainAccount(crossDomainAdmin) {
+    function setDepositQuoteTimeBuffer(uint64 buffer)
+        public
+        override
+        onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
+    {
         _setDepositQuoteTimeBuffer(buffer);
     }
 
     function initializeRelayerRefund(bytes32 relayerRepaymentDistributionProof)
         public
         override
         onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
     {
         _initializeRelayerRefund(relayerRepaymentDistributionProof);
     }
 
-    /**************************************
-     *         INTERNAL FUNCTIONS         *
-     **************************************/
-
-    function _setCrossDomainAdmin(address newCrossDomainAdmin) internal {
-        require(newCrossDomainAdmin != address(0), "Bad bridge router address");
-        crossDomainAdmin = newCrossDomainAdmin;
-        emit SetXDomainAdmin(crossDomainAdmin);
+    function _bridgeTokensToHubPool(MerkleLib.DestinationDistribution memory distributionLeaf) internal override {
+        IL2ERC20Bridge(Lib_PredeployAddresses.L2_STANDARD_BRIDGE).withdrawTo(
+            distributionLeaf.l2TokenAddress, // _l2Token. Address of the L2 token to bridge over.
+            hubPool, // _to. Withdraw, over the bridge, to the l1 pool contract.
+            distributionLeaf.amountToReturn, // _amount. Send the full balance of the deposit box to bridge.
+            6_000_000, // _l1Gas. Unused, but included for potential forward compatibility considerations
+            "" // _data. We don't need to send any data for the bridging action.
+        );
+        emit TokensBridged(
+            distributionLeaf.l2TokenAddress,
+            hubPool,
+            distributionLeaf.amountToReturn,
+            6_000_000,
```

Comment:
I think we should set this in a variable and then let the owner change it. does not need to happen in this PR though but should probs do it.

Reply:
I can change in this PR. what's annoying is that this is potentially Optimism specific, but HubPool will need to implement a method that can trigger the cross chain admin function to make this call. Shold we just implement `setL2GasLimit` in the `SpokePool` and have it be useless for some L2s?

Reply:
And consider other constants such as `maxSubmissionCost` and `l2GasPrice` (and `refundAddress`) that will be Arbitrum specific?

Reply:
Perhaps in the HubPool we can allow it to call any arbitrary encoded method to the SpokePool so it can call L2-specific admin functions like `setL2GasLimit` for Optimism and `setL2GasPrice` for Arbitrum

Reply:
ye, so I think that what makes the most sence is to have this be a chain specific event, like you mention. then, each SpokePool specific implementation for a given chain has these extra methods on them for those chain specific things.

Reply:
Cool added a universal event emitted by `SpokePool` and added an Optimism specific `OptimismTokenBridged` event. Also added a `setL1Gas onlyOwner` method on the OptimismSpokePool